packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -140,7 +140,7 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
     private attachListeners() {
         // Only listen to these events if not attached.
         if (!this.isAttached()) {
-            this.runtime.once("attaching", () => {
+            this.runtime.once("attached", () => {
```

Comment:
I think you are changing the meaning of this event as well (have not got that far).
And if so, I'd rather get rid of events altogether - they are pain in the but, so I'd rather switch to direct calls / APIs, and make it as symmetrical with data stores as possible.

That said, it looks like FFX wants us to keep back compat across data store and DDSs, so that may change how we approach this in the moment.